### DISCUSSION TITLE
How to refresh the git index when using `git_status_list` if the file has been updated but not modified?

### DISCUSSION THREAD

**Discussion by gmryuuko** [libgit2/libgit2#6852] [Q&A] [answered]
When the modification time of a file changes, but its content remains the same, calling the command line `git status` will update the modification time in the `.git/index` file. The next time `git status` is called, it will no longer need to perform the comparison.

If using `git_status_list` in libgit2, the modification time in the `.git/index` file is not updated. This leads to lengthy processing times as it requires repeated comparisons with the content of the working directory files.

Is there any way to update the `index` file while executing `git_status_list`?

**Comment by ethomson** [comment, marked as answer]
Yes - by default `status` doesn't update the index - if you pass `GIT_STATUS_OPT_UPDATE_INDEX` as a status option, it should do that update.

**Reply by ethomson** [reply]
(Note: this default is probably wrong, the default should probably be to write the index during diff and status operations.)

**Reply by gmryuuko** [reply]
Thank you for your reply! It works fine now.
